lib/icinga/notification.hpp

```diff
@@ -55,6 +55,7 @@ class ApplyRule;
 struct ScriptFrame;
 class Host;
 class Service;
+class UserGroup;
```

Comment:
This forward declaration isn't necessary!

Reply:
Doesn't work on my machine:

```
[36/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/user.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/user.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/user.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/user.cpp
FAILED: lib/icinga/CMakeFiles/icinga.dir/user.cpp.o 
/Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/user.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/user.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/user.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/user.cpp
In file included from /Users/aklimov/NET/WS/icinga2/lib/icinga/user.cpp:4:
In file included from /Users/aklimov/NET/WS/icinga2/build-clion/lib/icinga/user-ti.cpp:17:
In file included from /Users/aklimov/NET/WS/icinga2/lib/icinga/usergroup.hpp:7:
/Users/aklimov/NET/WS/icinga2/lib/icinga/notification.hpp:77:25: error: use of undeclared identifier 'UserGroup'
   77 |         std::set<intrusive_ptr<UserGroup>> GetUserGroups() const;
      |                                ^
1 error generated.
[37/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/scheduleddowntime.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/scheduleddowntime.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/scheduleddowntime.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/scheduleddowntime.cpp
[38/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/servicegroup.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/servicegroup.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/servicegroup.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/servicegroup.cpp
[39/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/service.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/service.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/service.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/service.cpp
[40/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/usergroup.cpp
FAILED: lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o 
/Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o -MF lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o.d -o lib/icinga/CMakeFiles/icinga.dir/usergroup.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/icinga/usergroup.cpp
In file included from /Users/aklimov/NET/WS/icinga2/lib/icinga/usergroup.cpp:3:
In file included from /Users/aklimov/NET/WS/icinga2/lib/icinga/usergroup.hpp:7:
/Users/aklimov/NET/WS/icinga2/lib/icinga/notification.hpp:77:25: error: use of undeclared identifier 'UserGroup'
   77 |         std::set<intrusive_ptr<UserGroup>> GetUserGroups() const;
      |                                ^
1 error generated.
[41/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/cli/CMakeFiles/cli.dir/pkiverifycommand.cpp.o -MF lib/cli/CMakeFiles/cli.dir/pkiverifycommand.cpp.o.d -o lib/cli/CMakeFiles/cli.dir/pkiverifycommand.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/cli/pkiverifycommand.cpp
[42/98] /Library/Developer/CommandLineTools/usr/bin/c++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/Users/aklimov/NET/WS/boost_1_69_0 -I/Users/aklimov/NET/WS/openssl/include -I/Users/aklimov/NET/WS/icinga2/third-party/nlohmann_json -I/Users/aklimov/NET/WS/icinga2/third-party/utf8cpp/source -I/Users/aklimov/NET/WS/icinga2 -I/Users/aklimov/NET/WS/icinga2/lib -I/Users/aklimov/NET/WS/icinga2/build-clion -I/Users/aklimov/NET/WS/icinga2/build-clion/lib -DI2_DEBUG=1 -Wall -Wextra -Wno-enum-constexpr-conversion -w -Qunused-arguments -fcolor-diagnostics -fno-limit-debug-info -pthread -Winvalid-pch -g -DI2_DEBUG -std=c++17 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk -mmacosx-version-min=14.7 -fcolor-diagnostics -MD -MT lib/cli/CMakeFiles/cli.dir/daemonutility.cpp.o -MF lib/cli/CMakeFiles/cli.dir/daemonutility.cpp.o.d -o lib/cli/CMakeFiles/cli.dir/daemonutility.cpp.o -c /Users/aklimov/NET/WS/icinga2/lib/cli/daemonutility.cpp
ninja: build stopped: subcommand failed.
```